x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserMlNodeServiceSettings.java

```diff
@@ -87,10 +87,21 @@ public ElserMlNodeServiceSettings(int numAllocations, int numThreads, String var
     public ElserMlNodeServiceSettings(StreamInput in) throws IOException {
         numAllocations = in.readVInt();
         numThreads = in.readVInt();
-        if (in.getTransportVersion().onOrAfter(TransportVersions.ELSER_SERVICE_MODEL_VERSION_ADDED)) {
+        if (transportVersionIsCompatibleWithElserModelVersion(in.getTransportVersion())) {
             modelVariant = in.readString();
         } else {
-            modelVariant = ElserMlNodeService.ELSER_V1_MODEL;
+            modelVariant = ElserMlNodeService.ELSER_V2_MODEL;
```

Comment:
I think it _would_ be better to stick to `ELSER_V1_MODEL` in the mixed version cluster where some of the nodes are pre-8.11.0. The `ELSER_V2_MODEL` states that it requires 8.11.0 (because it uses the platform specific config functionality). So it might not install in a mixed version cluster where some nodes pre-date 8.11.0.

Reply:
`ELSER_V1_MODEL` might not be available in the mixed cluster either if the upgrade is from pre-8.8. V2 is a better model and should be used in preference to V1.

There is validation to ensure that `modelVariant` is set when `ElserMlNodeServiceSettings` are PUT, this should mean that `modelVariant` is non null. In theory the branch setting the default value should never be hit for users of 8.11

Reply:
> In theory the branch setting the default value should never be hit for users of 8.11

Yes, true, and since this code doesn't exist in prior versions it should never be hit for mixed 8.11/earlier clusters either (at least for people running GA builds).

So I agree that what goes here doesn't matter, so we might as well leave it as-is.